firefox-ios/firefox-ios-tests/Tests/ClientTests/RustSyncManagerTests.swift

```diff
@@ -288,19 +321,23 @@ class LegacyRustSyncManagerTests: XCTestCase {
             .addresses
         ]
 
-        rustSyncManager.getEnginesAndKeys(engines: engines) { (engines, keys) in
```

Comment:
@yoanarios For some context, this test file is broken into `RustSyncManager` and `LegacySyncManagerTests`, the difference is enabling the feature flags. Legacy should be removed in the future. For `RustSyncManager` tests, I didn't add the `MockDispatchGroup()` since it needs to run through the actual dispatch group logic. The mock executes the completion right away instead of going through the dispatch group flow, so we need it there. Let me know if any concerns or improvements!